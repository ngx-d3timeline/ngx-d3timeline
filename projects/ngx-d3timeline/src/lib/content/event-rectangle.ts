import { identifier } from '../types';

export interface EventRectangle {
  id: identifier;
  title: string;
  transform: string;
  width: number;
  height: number;
}
