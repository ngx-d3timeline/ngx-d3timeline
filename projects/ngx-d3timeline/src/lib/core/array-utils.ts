export function sum(...args: number[]) {
  return args.reduce((a, b) => a + b);
}
