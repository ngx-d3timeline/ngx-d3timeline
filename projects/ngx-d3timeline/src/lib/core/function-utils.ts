export function partialApply<TInput, TAppliedArgs extends any[], TOut>(
  func: (input: TInput, ...appliedArgs: TAppliedArgs) => TOut
): (...appliedArgs: TAppliedArgs) => (input: TInput) => TOut {
  return (...appliedArgs: TAppliedArgs) => (input: TInput) =>
    func(input, ...appliedArgs);
}

export function pipe<T>(
  ...args: ((...args: any[]) => T)[]
): (...t: any[]) => T {
  return x => args.reduce((acc, curr) => curr(acc), x);
}

export function clampZero(x: number) {
  return Math.max(x, 0);
}
