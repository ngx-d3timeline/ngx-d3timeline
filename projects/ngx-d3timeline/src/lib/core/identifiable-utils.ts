import { Identifiable, identifier } from './types';

export function findIdentifiable<T extends Identifiable>(
  identifiables: T[],
  id: identifier
): T {
  return id && identifiables.find(identifiable => identifiable.id === id);
}
