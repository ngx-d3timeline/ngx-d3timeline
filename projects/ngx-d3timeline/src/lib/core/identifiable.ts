export type Identifier = string | number;

export interface Identifiable {
  id: Identifier;
}

export function findIdentifiable<T extends Identifiable>(
  identifiables: T[],
  id: Identifier
): T {
  return id && identifiables.find(identifiable => identifiable.id === id);
}
