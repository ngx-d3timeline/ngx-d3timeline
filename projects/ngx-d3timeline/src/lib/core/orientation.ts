export enum Orientation {
  Horizontal = 'Horizontal',
  Vertical = 'Vertical'
}

export function flipOrientation(orientation: Orientation): Orientation {
  return orientation === Orientation.Vertical
    ? Orientation.Horizontal
    : Orientation.Vertical;
}
