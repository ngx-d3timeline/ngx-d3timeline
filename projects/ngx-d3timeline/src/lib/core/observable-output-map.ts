import { Observable, Subject } from 'rxjs';
import { EventEmitter } from '@angular/core';
import { takeUntil } from 'rxjs/operators';

export interface ObservableOutputMap<T> {
  observable$: Observable<T>;
  output: EventEmitter<T>;
}

export function setupObservableToOutputMappings(
  observableToOutputMap: ObservableOutputMap<any>[],
  destroy$: Subject<boolean>
) {
  observableToOutputMap.forEach(map =>
    setupObservableToOutputMapping(map, destroy$)
  );
}

function setupObservableToOutputMapping<T>(
  observableOutputMap: ObservableOutputMap<T>,
  destroy$: Subject<boolean>
) {
  const { observable$, output } = observableOutputMap;
  observable$.pipe(takeUntil(destroy$)).subscribe(value => output.emit(value));
}
