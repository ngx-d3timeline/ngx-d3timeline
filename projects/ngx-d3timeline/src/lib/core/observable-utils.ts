import { Observable } from 'rxjs';
import { EventEmitter } from '@angular/core';
import { takeUntil } from 'rxjs/operators';

export function outputOnObservableEmit<T>(
  observable$: Observable<T>,
  takeUntil$: Observable<any>,
  output: EventEmitter<T>
) {
  observable$
    .pipe(takeUntil(takeUntil$))
    .subscribe(value => output.emit(value));
}
