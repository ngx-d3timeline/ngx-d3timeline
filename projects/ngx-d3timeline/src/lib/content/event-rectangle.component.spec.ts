import { TestBed, ComponentFixture } from '@angular/core/testing';
import { EventRectangleComponent } from './event-rectangle.component';

describe('EventRectangleComponent', () => {
  let fixture: ComponentFixture<EventRectangleComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [EventRectangleComponent]
    });

    fixture = TestBed.createComponent(EventRectangleComponent);
  });

  it('should not render if eventRectangle is null', () => {
    fixture.componentInstance.eventRectangle = null;

    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render correctly', () => {
    fixture.componentInstance.eventRectangle = {
      id: 1,
      title: 'Event 1',
      transform: 'translate(50,0)',
      width: 50,
      height: 80
    };

    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
