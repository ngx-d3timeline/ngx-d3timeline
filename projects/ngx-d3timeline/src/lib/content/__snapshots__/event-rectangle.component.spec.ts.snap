// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventRectangleComponent should not render if eventRectangle is null 1`] = `
<div
  id="root0"
  ngx-d3timeline-event-rectangle=""
>
  
</div>
`;

exports[`EventRectangleComponent should render correctly 1`] = `
<div
  id="root1"
  ngx-d3timeline-event-rectangle=""
>
  
  <g
    class="event-rectangle"
    transform="translate(50,0)"
  >
    <rect
      height="80"
      width="50"
    />
    <text
      dy="1em"
    >
      Event 1
    </text>
  </g>
</div>
`;
